src/rgw/rgw_amqp.cc

```diff
@@ -793,7 +805,7 @@ class Manager {
           ldout(cct, 10) << "AMQP run: unsolicited n/ack received with tag=" << tag << dendl;
         }
         // just increment the iterator
-        ++conn_it;
+        INCREMENT(conn_it);
```

Comment:
if this addresses a racing, shall we create a tracker, and backport this fix accordingly?

Reply:
Make sense, overall race condition is obviously happened. Added ticket info in commit message.